### PR TITLE
Fix DirectiveGenerator support of nested describables / improve testing

### PR DESCRIPTION
**Pull request by Vlatombe**
* Fix DirectiveGenerator support of nested describables

Downstream of https://github.com/jenkinsci/workflow-cps-plugin/pull/362

### CODE REVIEW

File: pipeline-model-definition/pom.xml [hunk 1]
```diff
@@ -112,6 +112,11 @@
       <groupId>org.jenkins-ci.plugins.workflow</groupId>
       <artifactId>workflow-cps</artifactId>
     </dependency>
+    <dependency>
+      <groupId>org.jenkins-ci.plugins.workflow</groupId>
+      <artifactId>workflow-cps</artifactId>
+      <classifier>tests</classifier>
```
Comment (dwnusbaum) [inline pipeline-model-definition/pom.xml:118]: I think this should be in test scope, right? It looks like there is a section below for test scope dependencies, so maybe better to move this down there.

```suggestion
      <classifier>tests</classifier>
      <scope>test</scope>
```

File: pipeline-model-definition/src/main/java/org/jenkinsci/plugins/pipeline/modeldefinition/generator/DirectiveGenerator.java [hunk 2]
```diff
@@ -107,10 +107,8 @@ public static String mapToClosure(Map<String,?> args) {
                 result.append(arg.getKey()).append(" ");
                 if (arg.getValue() instanceof Map) {
                     result.append(mapToClosure((Map<String, ?>) arg.getValue()));
-                } else if (arg.getValue() instanceof UninstantiatedDescribable) {
-                    result.append(mapToClosure(((UninstantiatedDescribable) arg.getValue()).getArguments()));
                 } else if (arg.getValue() != null) {
-                    result.append(Snippetizer.object2Groovy(arg.getValue())).append("\n");
+                    result.append(Snippetizer.object2Groovy(arg.getValue(), arg.getValue() instanceof UninstantiatedDescribable)).append("\n");
```
Comment (jglick) [inline pipeline-model-definition/src/main/java/org/jenkinsci/plugins/pipeline/modeldefinition/generator/DirectiveGenerator.java:111]: So is this changing behavior? If so, the PR title is misleading.

Comment (Vlatombe) [reply pipeline-model-definition/src/main/java/org/jenkinsci/plugins/pipeline/modeldefinition/generator/DirectiveGenerator.java:111]: Updated

File: pom.xml [hunk 3]
```diff
@@ -341,8 +347,9 @@
     <java.level>8</java.level>
     <groovy.version>2.4.12</groovy.version>
     <argLine>-Djava.awt.headless=true -Xmx4g -Xms512m -XX:MaxPermSize=512m</argLine>
-    <workflow-support-plugin.version>3.3</workflow-support-plugin.version>
-    <scm-api.version>2.4.1</scm-api.version>
+    <workflow-support-plugin.version>3.4</workflow-support-plugin.version>
+    <scm-api.version>2.6.3</scm-api.version>
+    <workflow-cps.version>2.81-rc2492.83316f41dee4</workflow-cps.version><!-- https://github.com/jenkinsci/workflow-cps-plugin/pull/362 -->
```
Comment (Vlatombe) [inline pom.xml:352]:
```suggestion
    <workflow-cps.version>2.81</workflow-cps.version>
```

**Comment by Vlatombe** [comment]
@dwnusbaum this should be ready to merge now.

**Comment by bitwiseman** [comment]
@Vlatombe  Could you do resolve the conflicts on this?

**Comment by Vlatombe** [comment]
@bitwiseman done

**Comment by bitwiseman** [comment]
@Vlatombe Looks like that failed compilation.

**Comment by Vlatombe** [comment]
@bitwiseman fixed

**Pull request merged**